src/slurm_plugin/slurm_resources.py

```diff
@@ -165,13 +166,14 @@ def is_drained(self):
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return any(
-            powering_down_state in self.states for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES
+        return (
+            any(powering_down_state in self.states for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES)
+            and self.SLURM_SCONTROL_POWERED_STATE not in self.states
         )
 
     def is_power(self):
         """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.states
+        return all(power_state in self.states for power_state in self.SLURM_SCONTROL_POWER_STATE)
```

Comment:
is this condition safe? I'm wondering if you could find the node in a state that is "IDLE", "CLOUD", "POWERED_DOWN" + something else which is not POWER_DOWN.

Shall we add the explicit states to SLURM_SCONTROL_POWER_STATE ({{"IDLE", "CLOUD", "POWERED_DOWN"},{"IDLE", "CLOUD", "POWERED_DOWN", "POWER_DOWN"}}) and check something like `self.states in self.SLURM_SCONTROL_POWER_STATE`?

Also can we add a comment on top of SLURM_SCONTROL_POWERED_STATE saying something like: "Due to a bug in Slurm a powered down node can enter IDLE+CLOUD+POWER_DOWN+POWERED_DOWN state"

Reply:
Yes, I think it is more save to use [{"IDLE", "CLOUD", "POWERED_DOWN"},{"IDLE", "CLOUD", "POWERED_DOWN", "POWER_DOWN"}], changed